### PR TITLE
special links/invalid URIs

### PR DESCRIPTION
**Pull request by tzarebczan**
Implements https://github.com/lbryio/lbry-app/issues/1352 - tested on Windows/Mac/Linux
Fixes https://github.com/lbryio/lbry-app/issues/959

This will also enable us to use magic links on Windows, as before the `?` was not being escaped properly. 
 Will need to update: https://github.com/lbryio/lbry.io/blob/b81191c7c5312183c25d072f3068b09fcdd1719e/view/template/acquisition/verify.php#L54

Tested magic link on Windows by setting the user agent. 

Thanks to @seanyesmunt for help with the doNotify logic :)

### CODE REVIEW

File: src/renderer/index.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ import app from './app';
 import analytics from './analytics';
 
 const { autoUpdater } = remote.require('electron-updater');
+const specialURL = 'lbry://?';
```
Comment (seanyesmunt) [inline src/renderer/index.js]: I think we should call this something else. Maybe `appPageUrl` or something. Just to give some indication of what it is

Comment (tzarebczan) [reply src/renderer/index.js]: good call.

File: src/renderer/component/router/view.jsx [hunk 2]
```diff
@@ -18,16 +18,21 @@ import InvitePage from 'page/invite';
...
 
-const route = (page, routesMap) => {
+const route = (props, page, routesMap) => {
   const component = routesMap[page];
-
-  return component || DiscoverPage;
+  if (!component) {
+        props.doNotify({
+        message: __('Invalid page requested'),
+        displayType: ['snackbar'],
```
Comment (seanyesmunt) [inline src/renderer/component/router/view.jsx]: can you try running `yarn format`? These two lines above should be indented once more.

File: src/renderer/index.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ import app from './app';
 import analytics from './analytics';
 
 const { autoUpdater } = remote.require('electron-updater');
+const appPageURL = 'lbry://?';
```
Comment (seanyesmunt) [inline src/renderer/index.js]: Generally constants like this are in all caps `APP_PAGE_URL`

Comment (tzarebczan) [reply src/renderer/index.js]: fixed

File: src/renderer/component/router/view.jsx (hunk 2 above)
Comment (tzarebczan) [reply src/renderer/component/router/view.jsx]: fixed this one, there may be some others from previous commits. If you want, run prettier on this branch please.

**Comment by seanyesmunt** [comment]
@tzarebczan I rebased and squashed your commits. I'll merge once it's green :slightly_smiling_face:

**Comment by seanyesmunt** [comment]
Not sure why the `push` travis build is looking at the old `.travis.yml`. The PR build passed so good to go.

**Comment by tzarebczan** [comment]
I didn't update the Travis config on this build, that's why.

**Pull request merged**